### PR TITLE
[Distributed] Companion fix after pytorch PR73164

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
Author: @esaliya

PyTorch PR pytorch/pytorch#73164 adds changes to extend the 3rd party
distributed backend registration. As part of this the internal data
structure is changed to a named tuple of {creator_fn, bool}. This
causes some of the tests in xla to fail as it expects a callable
being returned for a particular backend name.

This PR fixes this issue by calling the creator_fn of the namedtuple
instead of the namedtuple itself.

Test Plan:
python test/test_torch_distributed_xla_backend.py

### CODE REVIEW

**Comment by alanwaketan** [comment]
Cc @esaliya. This is the effort to land #3982.

**Review by JackCaoG — APPROVED** [review]

**Comment by kumpera** [comment]
Could we extend PTD's Backend to avoid this in the future?

**Comment by alanwaketan** [comment]
> Could we extend PTD's Backend API to avoid this in the future?

Avoid what?

File: torch_patches/.torch_pin [hunk 1]
```diff
@@ -0,0 +1 @@
+#73164
```
Comment (vanbasten23) [inline torch_patches/.torch_pin:1]: This means this PR works only with the change in https://github.com/pytorch/pytorch/pull/73164. Then would you check in this file when you merge this PR, or would you wait until https://github.com/pytorch/pytorch/pull/73164 merges first and then discard change in this torch_patches/.torch_pin?

Comment (alanwaketan) [reply torch_patches/.torch_pin:1]: I believe the Later. You can check out this guide for more details. https://github.com/pytorch/xla/blob/master/.circleci/README.md.

Comment (vanbasten23) [reply torch_patches/.torch_pin:1]: Thanks!

**Pull request merged**